src/Symfony/Component/Notifier/Bridge/ClickSend/ClickSendTransport.php

```diff
@@ -0,0 +1,127 @@
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author gnito-org <https://github.com/gnito-org>
+ */
+final class ClickSendTransport extends AbstractTransport
+{
+    protected const HOST = 'rest.clicksend.com';
+
+    public function __construct(
+        private readonly string $apiUsername,
+        #[\SensitiveParameter] private readonly string $apiKey,
+        private readonly ?string $from = null,
+        private readonly ?string $source = null,
+        private readonly ?string $listId = null,
+        private readonly ?string $fromEmail = null,
+        HttpClientInterface $client = null,
+        EventDispatcherInterface $dispatcher = null
+    ) {
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        $queryParameters = [];
+        if ($this->from) {
+            $queryParameters['from'] = $this->from;
+        }
+        if ($this->source) {
+            $queryParameters['source'] = $this->source;
+        }
+        if ($this->listId) {
+            $queryParameters['list_id'] = $this->listId;
+        }
+        if ($this->fromEmail) {
+            $queryParameters['from_email'] = $this->fromEmail;
+        }
+
+        return sprintf('clicksend://%s', $this->getEndpoint()).($queryParameters ? '?'.http_build_query($queryParameters) : null);
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof SmsMessage;
+    }
+
+    /**
+     * https://developers.clicksend.com/docs/rest/v3/#send-sms.
+     */
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof SmsMessage) {
+            throw new UnsupportedMessageTypeException(__CLASS__, SmsMessage::class, $message);
+        }
+
+        $endpoint = sprintf('https://%s/v3/sms/send', $this->getEndpoint());
+
+        $opts = $message->getOptions();
+        $options = $opts ? $opts->toArray() : [];
+        $options['body'] = $message->getSubject();
+
+        if (!isset($options['from']) && $this->from) {
+            $options['from'] = $this->from;
+        }
+
+        if (isset($options['from']) && !preg_match('/^[a-zA-Z0-9\s]{3,11}$/', $options['from']) && !preg_match('/^\+[1-9]\d{1,14}$/', $options['from'])) {
+            throw new InvalidArgumentException(sprintf('The "From" number "%s" is not a valid phone number, shortcode, or alphanumeric sender ID.', $this->from));
+        }
+
+        if (!isset($options['source']) && $this->source) {
+            $options['source'] = $this->source;
+        }
+
+        if (isset($options['list_id']) && $this->listId) {
+            $options['list_id'] = $this->listId;
+        }
+
+        if (isset($options['from_email']) && $this->fromEmail) {
+            $options['from_email'] = $this->fromEmail;
+            $options['from_email'] = urldecode($options['from_email']);
```

Comment:
```suggestion
            $options['from_email'] = urldecode($this->fromEmail);
```
? Or am I missing something? I mean its overridden immediately 🤔